test-workloads/src/bin/tellus.rs

```diff
@@ -344,6 +374,33 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
                         }
                         zero_pages_added += 1;
                     }
+                    sbi::TvmCpuExitCode::SharedPageFault => {
+                        // Figure out where the fault occurred.
+                        let fault_addr = get_fault_address(vmid);
+                        if fault_addr != SHARED_PAGES_START_ADDRESS {
+                            println!("Fault in shared page region at {fault_addr:x}");
+                            panic!("Unexpected shared page fault address");
+                        }
+                        let msg = SbiMessage::Tee(sbi::TeeFunction::TvmAddSharedPages {
+                            guest_id: vmid,
+                            page_addr: next_page,
+                            page_type: sbi::TsmPageType::Page4k,
+                            num_pages: 1,
+                            guest_addr: fault_addr,
+                        });
+                        // Safety: `TvmAddSharedPages` only touches pages owned by us
+                        unsafe {
+                            ecall_send(&msg).expect("Tellus -- TvmAddSharedPages failed");
+                        }
+
+                        // Safety: We own the page, and are writing a value expected by the guest
```

Comment:
This is giving up exclusive ownership of the page. It would be nice if the comment mentioned that and that tellus won't access it again, or must use volatile, shared memory access patterns.